clients/drcachesim/tests/view_test.cpp

```diff
@@ -329,14 +402,32 @@ run_limit_tests(void *drcontext)
         gen_branch(t1, offs_nop2),
         gen_data(t1, true, 0x42, 4),
     };
+    // To test skipping and limiting we need to use an analyzer which requires
+    // a trace_entry_t version of the above.
+    std::vector<trace_entry_t> records = {
```

Comment:
So we want this to be exactly the same as the `std::vector<memref_t>` above? May be hard to ensure that it always stays in sync. See question in other comment thread about whether we really need the `std::vector<memref_t>` anymore.

Side question: We have logic in reader_t that converts `trace_entry_t`s to`memref_t`s. Can that be reused somehow?